testing/tests/configs.py

```diff
@@ -392,3 +393,155 @@ def test_orphan_typedef(self):
     def test_orphan_file(self):
         """Verify excluding ``file`` behaves as expected."""
         self._validate_diff_for_kinds(1, "file")
+
+
+@confoverrides(exhale_args={
+    "exhaleDoxygenStdin": textwrap.dedent("""\
+        INPUT            = ../include
+        EXCLUDE_PATTERNS = */page_town_rock*.hpp
+    """)})
+class ManualIndexingTests(ExhaleTestCase):
+    """
+    Tests for :ref:`manual_indexing`.
+
+    Includes checks for:
+
+    - :data:`~exhale.configs.rootFileName`
+    - :data:`~exhale.configs.classHierarchyFilename`
+    - :data:`~exhale.configs.fileHierarchyFilename`
+    - :data:`~exhale.configs.pageHierarchyFilename`
+    - :data:`~exhale.configs.unabridgedApiFilename`
+    - :data:`~exhale.configs.unabridgedOrphanFilename`
+    """
+
+    test_project = "cpp_nesting"
+    """.. testproject:: cpp_nesting"""
+
+    def root_file_path(self) -> Path:
+        """Return the path to the ``rootFileName``."""
+        root_file_name = self.app.config.exhale_args["rootFileName"]
+        return Path(self.getAbsContainmentFolder()) / root_file_name
+
+    def verify_expected_files(
+        self,
+        *,
+        page_default: str = "page_view_hierarchy.rst.include",
+        class_default: str = "class_view_hierarchy.rst.include",
+        file_default: str = "file_view_hierarchy.rst.include",
+        u_api_default: str = "unabridged_api.rst.include",
+        u_orphan_default: str = "unabridged_orphan.rst"
+    ):
+        """Verify the expected files were generated and included."""
+        # The files that may or may not be generated by exhale.
+        containment_folder = Path(self.getAbsContainmentFolder())
+        root_path = Path(self.app.exhale_root.full_root_file_path)
+        page_path = Path(self.app.exhale_root.page_hierarchy_file)
+        class_path = Path(self.app.exhale_root.class_hierarchy_file)
+        file_path = Path(self.app.exhale_root.file_hierarchy_file)
+        u_api_path = Path(self.app.exhale_root.unabridged_api_file)
+        u_orphan_path = Path(self.app.exhale_root.unabridged_orphan_file)
+
+        # Calling tests change the exhale_args and corresponding default parameter,
+        # forcing manual effort deliberately for test correctness.
+        def verify_name(name: str, actual: Path):
+            # These are all in the same directory.
+            self.assertEqual(
+                containment_folder / name,
+                actual,
+                f"Expected {name} and {actual.name} to be the same."
+            )
```

Comment:
`E   AssertionError: WindowsPath('//?/D:/a/exhale/exhale/testing/projects/c[97 chars]ude') != WindowsPath('D:/a/exhale/exhale/testing/projects/cpp_n[93 chars]ude')`: `//?` magic prefix issues on windows.  issues on mac too it seems